tests/unit_tests/net.cpp

```diff
@@ -936,6 +936,35 @@ TEST(get_network_address, ipv4subnet)
     EXPECT_STREQ("12.34.0.0/16", address->str().c_str());
 }
 
+TEST(get_network_address, host_and_port)
+{
+#   define NA_HOST_AND_PORT_TEST(addr, exp_host, exp_port)            \
```

Comment:
This really should be a lambda. Its about the same number of characters and isn't hijacking the preprocessor. If the concern is over line numbers then `SCOPED_TRACE(addr);` can track what failed, but even then it should be obvious based on the `exp_host` and `exp_port` output.